### ISSUE TITLE
SIGSEGV when using syslog-ng in python interactive mode

### ISSUE THREAD

**Issue by mitzkia** [syslog-ng/syslog-ng#614]
Started syslog-ng with the following configuration:

```
@version: 3.7

source s_internal { internal(); };
source s_file { file("json.txt"); };

destination d_file {
  file("/tmp/output.log");
};

log {
  source(s_internal);
  source(s_file);
  destination(d_file);
};
```

syslog-ng start command: `sbin/syslog-ng -F -f etc/syslog-ng2.conf -i --enable-core`

steps in interactive mode:

```
Waiting for breakpoint...
Breakpoint hit etc/syslog-ng2.conf:3:21
3        source s_internal { internal(); };
Dec 31 23:59:58  syslog-ng[13050]: syslog-ng starting up; version='3.7.0beta2'
(syslog-ng) c
(continuing)
Breakpoint hit etc/syslog-ng2.conf:3:21
3        source s_internal { internal(); };
Aug  4 16:23:25 tristram syslog-ng[13050]: syslog-ng starting up; version='3.7.0beta2'
(syslog-ng) c
(continuing)
Breakpoint hit etc/syslog-ng2.conf:3:20
3        source s_internal { internal(); };
Aug  4 16:23:25 tristram syslog-ng[13050]: syslog-ng starting up; version='3.7.0beta2'
(syslog-ng) c
(continuing)
Breakpoint hit etc/syslog-ng2.conf:3:1
3        source s_internal { internal(); };
Aug  4 16:23:25 tristram syslog-ng[13050]: syslog-ng starting up; version='3.7.0beta2'
(syslog-ng) c
(continuing)
Breakpoint hit etc/syslog-ng2.conf:3:1
3        source s_internal { internal(); };
Aug  4 16:23:25 tristram syslog-ng[13050]: syslog-ng starting up; version='3.7.0beta2'
(syslog-ng) c
(continuing)
Breakpoint hit etc/syslog-ng2.conf:14:3
14         source(s_internal);
Aug  4 16:23:25 tristram syslog-ng[13050]: syslog-ng starting up; version='3.7.0beta2'
(syslog-ng) c
(continuing)
Breakpoint hit etc/syslog-ng2.conf:14:3
14         source(s_internal);
Aug  4 16:23:25 tristram syslog-ng[13050]: syslog-ng starting up; version='3.7.0beta2'
(syslog-ng) c
(continuing)
Breakpoint hit etc/syslog-ng2.conf:17:3
17         destination(d_file);
Aug  4 16:23:25 tristram syslog-ng[13050]: syslog-ng starting up; version='3.7.0beta2'
(syslog-ng) c
(continuing)
Breakpoint hit etc/syslog-ng2.conf:9:21
9        destination d_file {
Aug  4 16:23:25 tristram syslog-ng[13050]: syslog-ng starting up; version='3.7.0beta2'
(syslog-ng) c
(continuing)
Breakpoint hit etc/syslog-ng2.conf:10:3
10         file("/tmp/output.log");
Aug  4 16:23:25 tristram syslog-ng[13050]: syslog-ng starting up; version='3.7.0beta2'
(syslog-ng) c
(continuing)
Breakpoint hit etc/syslog-ng2.conf:10:3
10         file("/tmp/output.log");
Aug  4 16:23:25 tristram syslog-ng[13050]: syslog-ng starting up; version='3.7.0beta2'
(syslog-ng) c
(continuing)
Breakpoint hit etc/syslog-ng2.conf:10:3
10         file("/tmp/output.log");
Aug  4 16:23:25 tristram syslog-ng[13050]: syslog-ng starting up; version='3.7.0beta2'
(syslog-ng) c
(continuing)
Breakpoint hit etc/syslog-ng2.conf:9:21
9        destination d_file {
Aug  4 16:23:25 tristram syslog-ng[13050]: syslog-ng starting up; version='3.7.0beta2'
(syslog-ng) c
(continuing)
Breakpoint hit etc/syslog-ng2.conf:5:17
5        source s_file { file("json.txt"); };
Aug  4 16:23:32  {"widget": {
(syslog-ng) c
(continuing)
Breakpoint hit etc/syslog-ng2.conf:5:17
5        source s_file { file("json.txt"); };
Aug  4 16:23:32 tristram {"widget": {
(syslog-ng) q
(continuing)
Breakpoint hit etc/syslog-ng2.conf:5:17
5        source s_file { file("json.txt"); };
Dec 31 23:59:58  syslog-ng[13050]: syslog-ng shutting down; version='3.7.0beta2'
(syslog-ng) q
(continuing)
Breakpoint hit etc/syslog-ng2.conf:3:21
3        source s_internal { internal(); };
Aug  4 16:23:32  "window": {
(syslog-ng) q
(continuing)
Breakpoint hit etc/syslog-ng2.conf:5:17
5        source s_file { file("json.txt"); };
Segmentation fault (core dumped)
```

core backtrace:

```
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `sbin/syslog-ng -F -f etc/syslog-ng2.conf -i --enable-core'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f4ac35b7aa8 in log_macro_expand (result=0x7f4abc001ce0, id=22, escape=0, opts=0x1692b88, tz=0, seq_num=0, context_id=0x0, msg=0x0) at ../lib/template/macros.c:474
474         zone_ofs = (opts->time_zone_info[tz] != NULL ? time_zone_info_get_offset(opts->time_zone_info[tz], stamp->tv_sec) : stamp->zone_offset);
(gdb) bt full
#0  0x00007f4ac35b7aa8 in log_macro_expand (result=0x7f4abc001ce0, id=22, escape=0, opts=0x1692b88, tz=0, seq_num=0, context_id=0x0, msg=0x0) at ../lib/template/macros.c:474
        buf = "\000\000\000\000\000\000\000\000\177%\030\301J\177\000\000\377\377\377\377\000\000\000\000\217%\030\301J\177\000\000|\000\000\000J\177\000\000\340o\000\303J\177", '\000' <repeats 11 times>, "\250i\001\000\000\000"
        length = 32586
        stamp = 0x20
        sstamp = {tv_sec = 0, tv_usec = 0, zone_offset = 0}
        tmp_hour = 0
        tm = 0x0
        tm_storage = {tm_sec = 0, tm_min = 4, tm_hour = 48, tm_mday = 4, tm_mon = -1055391744, tm_year = 32586, tm_wday = 0, tm_yday = 0, tm_isdst = -1055382128, tm_gmtoff = 0, 
          tm_zone = 0xffffffffffffffff <error: Cannot access memory at address 0xffffffffffffffff>}
        t = 139959073059381
        zone_ofs = 139959043893072
        __func__ = "log_macro_expand"
#1  0x00007f4ac35b6473 in log_template_append_format_with_context (self=0x169a5b0, messages=0x7f4ac1182690, num_messages=1, opts=0x1692b88, tz=0, seq_num=0, context_id=0x0, result=0x7f4abc001ce0) at ../lib/template/templates.c:125
        len = 0
        msg_ndx = 0
        p = 0x16bb500
        e = 0x169a680
#2  0x00007f4ac35b66af in log_template_append_format (self=0x169a5b0, lm=0x0, opts=0x0, tz=0, seq_num=0, context_id=0x0, result=0x7f4abc001ce0) at ../lib/template/templates.c:177
No locals.
#3  0x00007f4ac35b6705 in log_template_format (self=0x169a5b0, lm=0x0, opts=0x0, tz=0, seq_num=0, context_id=0x0, result=0x7f4abc001ce0) at ../lib/template/templates.c:184
No locals.
#4  0x00007f4ac35bca11 in _display_msg_with_template (self=0x1684b70, msg=0x0, template=0x169a5b0) at ../lib/debugger/debugger.c:69
        output = 0x7f4abc001ce0
#5  0x00007f4ac35bd109 in _handle_interactive_prompt (self=0x1684b70) at ../lib/debugger/debugger.c:296
        buf = "etc/syslog-ng2.conf:5:17", '\000' <repeats 336 times>...
        current_pipe = 0x1695610
#6  0x00007f4ac35bd185 in _interactive_console_thread_func (self=0x1684b70) at ../lib/debugger/debugger.c:316
No locals.
#7  0x00007f4ac329af05 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#8  0x00007f4ac3017182 in start_thread (arg=0x7f4ac1185700) at pthread_create.c:312
        __res = <optimized out>
        pd = 0x7f4ac1185700
        now = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {139959043905280, 7632211822920913163, 0, 0, 139959043905984, 139959043905280, -7530420966325956341, -7530416368896257781}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, 
            data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
        not_first_call = <optimized out>
        pagesize_m1 = <optimized out>
        sp = <optimized out>
        freesize = <optimized out>
---Type <return> to continue, or q <return> to quit---
        __PRETTY_FUNCTION__ = "start_thread"
```

syslog-ng was builded from source (master), latest patch

```
commit 1ffaea02bb66e781889237923fc37173300fb715
Merge: 253a6a8 c0bf9fe
Author: MÓZES Ádám István <dnsjts@users.noreply.github.com>
Date:   Tue Aug 4 10:29:03 2015 +0200

    Merge pull request #611 from lbudai/f/java-http-dst-fix

    :+1:, thanks!
```

**Comment by ihrwein** [comment]
syslog-ng was in the middle of the shutdown process when it got an other `q` command.

Just my two cents: we should consider this as a wontfix or we may cache that we are shutting down.

**Comment by bazsi** [comment]
We shouldn't crash anyway.

**Comment by MrAnno** [comment]
Can't reproduce.
Probably fixed in #1537.
